src/Symfony/Component/Security/Http/EventListener/CsrfTokenClearingLogoutListener.php

```diff
@@ -31,6 +32,10 @@ public function __construct(ClearableTokenStorageInterface $csrfTokenStorage)
 
     public function onLogout(LogoutEvent $event): void
     {
+        if ($this->csrfTokenStorage instanceof SessionTokenStorage && !$event->getRequest()->hasPreviousSession()) {
```

Comment:
What about fixing it in `SessionTokenStorage` instead?

Reply:
Using `SessionTokenStorage` without a session has been deprecated in 5.x: https://github.com/symfony/symfony/blob/8c637a5bbfea8b322ec93eb51ac47c0db8948e2a/src/Symfony/Component/Security/Csrf/TokenStorage/SessionTokenStorage.php#L95-L108

Ideally this listener shouldn't be registered for stateless firewalls, problem is that it's not a per-firewall listener but a global one. We should probably change that in another (feature) PR.